### ISSUE TITLE
API

### ISSUE THREAD

**Issue by ijlyttle** [ijlyttle/btt22#2]
I can see two exported functions, but I need to noodle on how to design the formals:

```r
# moves templating files into `R/` and/or `tests/testthat`
btt_get <- function()

# completely wipes out `R/` and `tests/testthat`, then populates
btt_reset <- function()
```

I'd like the argument(s):
- to describe a point in time during the workshop
- to be the same for both functions

That way, if a student is on the happy path, they could `btt_get()` for a point in time to get the next batch of files.
If a student is not on the happy path, they could `btt_reset()` for the same point in time, but all their previous work would be replaced.

**Comment by ijlyttle** [comment]
We would also have to manage the `Imports` and `Suggests` sections of the `DESCRIPTION` file.

**Comment by ijlyttle** [comment]
The argument could be *version*; we could use [`numeric_version()`](https://stat.ethz.ch/R-manual/R-devel/library/base/html/numeric_version.html).

We could have a schema where the major version is the day (`1`, `2`), the minor version is the session (`1`, `2`, `3`, `4`) and the patch version could identify a step within a session.

```r
btt_get("2.1.1")
```

**Comment by ijlyttle** [comment]
This is starting to sound a little like git, where a version number could be a tag.

**Comment by ijlyttle** [comment]
Thinking about it more, I think we should use branches rather than tags.
Tags are meant to be permanent; we instructors will need to be flexible.

Let's say we have a set of branches `2.1.1`, `2.2.1`, etc. the branch `2.1.1` would represent the starting point for day 2 - the all-important interface.

Let's say Ian makes a first idea of what `R`, `tests/testthat`, and `DESCRIPTION` (parts) might look like, but that Emma arrives at a slightly different place.

In this case, Emma could make a PR into the `2.1.1` branch, then once it is integrated, Ian could merge those changes into the subsequent branches.

**Comment by ijlyttle** [comment]
I know I'm probably running off at the the mouth (well, keyboard), but the most-precious commodity we have is those two days where we are all in the classroom. This may impose on us (instructors) in the leadup to the course, but if we can have a robust way for a student to "pull the ripcord on the parachute" to land back in the happy path (my gosh, the mixed metaphors), I'd like to find that way.

Hopefully this splashes mostly on day 2 so I would just be imposing on myself...

**Comment by ijlyttle** [comment]
Thought about it a bit more, I think:

```r
btt_reset("2.1.1") 
```

should get you all the *previous* files in `R`, `tests`, and packages in `DESCRIPTION`.

I think 

```r
btt_get("2.1.1")
```

should get you all the templated files for the upcoming exercise.

In other words, the actions would be independent.

This way, if someone falls off the happy path, they can use `btt_reset()` to get caught up, run R CMD CHECK to reassure themselves, then use `btt_get()` to get the templated files (which will likely not work).

**Comment by ijlyttle** [comment]
implemented (largely) in #5 and #7
